### PR TITLE
Create doctors controller, actions for the endpoints and test them

### LINKED ISSUE #8: [3 pts] Create a Doctors Controller - Member 1
- Create the following methods for the controller

`index`
`show`
`new`
`create`
`destroy`

- Configure routes for all the methods

### PR DESCRIPTION
**Pull request by JonahKayizzi**
This pull request consists of the following;
- Configure routes for the controllers endpoints
- Create the following endpoints
`http://127.0.0.1:3000/api/v1/doctors` - index
`http://127.0.0.1:3000/api/v1/doctors/:id` - show
POST `http://127.0.0.1:3000/api/v1/doctors` - create
DELETE `http://127.0.0.1:3000/api/v1/doctors/:id` - destroy
- Created the following actions for the controller endpoints
`index`
`show`
`new`
`create`
`destroy`
- Created specs for the endpoints

### CODE REVIEW

File: app/controllers/api/v1/doctors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  def show
+    @doctor = Doctor.find(params[:id])
+    render json: @doctor
+  end
+
+  def create
+    @doctor = Doctor.new(doctor_params)
+    if @doctor.save
+      render json: 'Doctor created successfully'
```
Comment (cvalencia1991) [inline app/controllers/api/v1/doctors_controller.rb:15]: you should implement this way of the controller like this [link](https://web-crunch.com/posts/create-a-basic-api-with-ruby-on-rails),also, you should implement it like the next text:
```
def create 
  @band = Band.new(band_params)
  if @band.save
        render json: @band, status: :created, location: @band
      else
        render json: @band.errors, status: :unprocessable_entity
      end
end
```

File: app/controllers/api/v1/doctors_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    else
+      render json: 'Doctor creation failed'
+    end
+  end
+
+  def doctor_params
+    params.require(:doctor).permit(:name, :speciality, :description, :graduation, :image)
+  end
```
Comment (cvalencia1991) [inline app/controllers/api/v1/doctors_controller.rb]: you should put this method in private, is exclusive for this class

**Review by cvalencia1991 — CHANGES_REQUESTED** [review]
##
## Hi @JonahKayizzi  ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

Great Job implementing The API, but still, have changes to be implemented.

_Check the comments under the review._

Cheers and Happy coding!👏👏👏

File: app/controllers/api/v1/doctors_controller.rb (hunk 1 above)
Comment (JonahKayizzi) [reply app/controllers/api/v1/doctors_controller.rb:15]: I do not think there is a problem with this implementation

File: app/controllers/api/v1/doctors_controller.rb (hunk 2 above)
Comment (JonahKayizzi) [reply app/controllers/api/v1/doctors_controller.rb]: @cvalencia1991 , the doctor_param method has been made private.

**Review by cvalencia1991 — APPROVED** [review]

**Pull request merged**